fc/mef.py

```diff
@@ -736,230 +506,223 @@ def get_transform_fxn(data_beads, peaks_mef, mef_channels,
+        populations_channel = [pi[:, mef_channel] for pi in populations]
+
+        ###
+        # 2. Calculate statistics in each subpopulation.
+        ###
+
+        # Calculate statistics
+        stats_values = [statistic_func(pi, **statistic_params)
+                        for pi in populations_channel]
+        stats_values = np.array(stats_values)
+
+        # Accumulate results
+        if full_output:
+            stats_values_all.append(stats_values)
 
-        # Slice relevant channel's data
-        data_channel = data_beads[:,mef_channel]
         # Print information
-        if verbose: 
-            print("- MEF transformation for channel {}...".format(mef_channel))
-
-        # Step 2. Find peaks in each one of the clusters. 
-        # ===============================================
-
-        # Find peaks on all the channel data
-        min_fl = data_channel.domain(0)[0]
-        max_fl = data_channel.domain(0)[-1]
-        if find_peaks_method == 'smoothed_mode':
-            # Set default values for limit values
-            if 'min_val' not in find_peaks_params:
-                find_peaks_params['min_val'] = min_fl
-            if 'max_val' not in find_peaks_params:
-                find_peaks_params['max_val'] = max_fl
-            # Get peak values
-            peaks_hists = [find_peaks_smoothed_mode(di[:,mef_channel], 
-                                                        **find_peaks_params)
-                                    for di in data_clustered]
-            peaks_ch = np.array([ph[0] for ph in peaks_hists])
-            hists_smooth = [ph[1] for ph in peaks_hists]
-        elif find_peaks_method == 'median':
-            peaks_ch = np.array([find_peaks_median(di[:,mef_channel],
-                                                        **find_peaks_params)
-                                    for di in data_clustered])
+        if verbose:
+            print("({}) Step 2: Population Statistic".format(mef_channel))
+            print("  Fluorescence per population (Channel Units):")
+            print("    " + str(stats_values))
+
+        ###
+        # 3. Select populations to be used for fitting
+        ###
+
+        # Select populations based on selection_func
+        if selection_func is not None:
+            selected_mask = selection_func([pi for pi in populations_channel],
+                                           **selection_params)
         else:
-            raise ValueError("Peak finding method {} not recognized."
-                .format(find_peaks_method))
-        # Sort peaks and clusters
-        ind_sorted = np.argsort(peaks_ch)
-        peaks_sorted = peaks_ch[ind_sorted]
-        data_sorted = [data_clustered[i] for i in ind_sorted]
+            selected_mask = np.ones(n_clusters, dtype=bool)
+
+        # Discard values specified as nan in mef_values_channel
+        selected_mask = np.logical_and(selected_mask,
+                                       ~np.isnan(mef_values_channel))
```

Comment:
As mentioned above, not sure why you went with masks here when you fought so hard against them in `gate` module. `selection_func` could just as easily return `populations` already filtered by the mask.

Also used to seeing `&`. As mentioned above, any reason for `np.logical_and` over `&`?